creusot-contracts/src/lib.rs

```diff
@@ -27,6 +27,14 @@ mod macros {
 
     pub use base_macros::snapshot;
 
+    /// Indicate that the function terminates: fullfilling the `requires` clauses
+    /// ensures that this function will not loop indefinitively.
+    pub use base_macros::terminates;
+
+    /// Indicate that the function is pure: fullfilling the `requires` clauses ensures
+    /// that this function will terminate, and will not panic.
```

Comment:
I think this description is a bit unclear since Creusot forces to you to prove absence of panics. I think we need to clarify that for programs Creusot actually 'allows' some kinds of panics (ie overflow of `Vec::len`). Otherwise it looks like `terminates = pure`. 

I wonder if it could be interesting to have programs which _do not_ verify the absence of panics? where the contract then says "if the function terminates successfully, _then_ the result satisfies the postcondition? (Not for here / now at all)

Reply:
Indeed, I added lines about this just below this doc